### PR TITLE
Add CI for all supported platforms

### PR DESCRIPTION
**Pull request by thebevrishot**
## PR intention
Resolve the problem that we fail to build target Windows and mac OS when we build release. This PR will enable GitHub actions to build target all supported platforms when master is change.

## Code changes brief
This script has 3 separated jobs that build target Linux(run unit tests and RPC tests), Windows and mac OS

Closes https://github.com/zcoinofficial/zcoin/issues/766

### CODE REVIEW

File: .github/workflows/ci-master.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+    - name: Install dependencies
+      run: |
+        sudo apt update
+        sudo apt-get install -y ccache python3-zmq
+    - name: Build dependencies
+      run: make -C depends -j2 HOST=x86_64-unknown-linux-gnu
```
Comment (ultimaweapon) [inline .github/workflows/ci-master.yml]: I think we should use `-j$(nproc)` instead.

File: .github/workflows/ci-master.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    - name: Build Zcoin
+      run: |
+        ./autogen.sh
+        ./configure --enable-exodus --enable-tests --with-comparison-tool=no --prefix=$(pwd)/depends/x86_64-unknown-linux-gnu
+        make -j2
```
Comment (ultimaweapon) [inline .github/workflows/ci-master.yml]: Same here.

File: .github/workflows/ci-master.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    name: Target Windows
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+    - name: Install dependencies
+      run: |
+        sudo apt update
+        sudo apt install -y ccache g++-mingw-w64-x86-64 mingw-w64-x86-64-dev
+    - name: Update gcc mingw
```
Comment (ultimaweapon) [inline .github/workflows/ci-master.yml]: This not quite correct. My idea is `Switch MinGW GCC to POSIX Threading`.

File: .github/workflows/ci-master.yml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      run: sudo update-alternatives --set x86_64-w64-mingw32-gcc /usr/bin/x86_64-w64-mingw32-gcc-posix
+    - name: Update g++ mingw
+      run: sudo update-alternatives --set x86_64-w64-mingw32-g++ /usr/bin/x86_64-w64-mingw32-g++-posix
```
Comment (ultimaweapon) [inline .github/workflows/ci-master.yml]: I think better to merge this one with the above one once we updated it label.

File: .github/workflows/ci-master.yml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    - name: Build dependencies
+      run: make -C depends -j2 HOST=x86_64-w64-mingw32
```
Comment (ultimaweapon) [inline .github/workflows/ci-master.yml]: Same as previous issue.

File: .github/workflows/ci-master.yml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    - name: Build Zcoin
+      run: |
+        ./autogen.sh
+        ./configure --disable-jni --enable-exodus --prefix=$(pwd)/depends/x86_64-w64-mingw32
+        make -j2
```
Comment (ultimaweapon) [inline .github/workflows/ci-master.yml]: Same here.

File: .github/workflows/ci-master.yml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    - name: Checkout
+      uses: actions/checkout@v1
+    - name: Install dependencies
+      run: |
+        sudo apt update
+        sudo apt-get install -y ccache libcap-dev
+    - name: Prepare directory for SDKs
+      run: mkdir -p depends/SDKs
+    - name: Download SDK
+      run: curl -L -o /tmp/MacOSX10.11.sdk.tar.gz https://bitcoincore.org/depends-sources/sdks/MacOSX10.11.sdk.tar.gz
```
Comment (ultimaweapon) [inline .github/workflows/ci-master.yml]: We should not do this. We should host this by ourself. But the problems with that way is it still violate macOS SDK license. So my idea is we should build on macOS directly so we don't need this SDK.

File: .github/workflows/ci-master.yml [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    - name: Extract SDK
+      run: tar -C depends/SDKs -xf /tmp/MacOSX10.11.sdk.tar.gz
+    - name: Build dependencies
+      run: make -C depends -j2 HOST=x86_64-apple-darwin11
```
Comment (ultimaweapon) [inline .github/workflows/ci-master.yml]: Same as previous issue.

File: .github/workflows/ci-master.yml [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    - name: Build Zcoin
+      run: |
+        ./autogen.sh
+        ./configure --enable-exodus --prefix=$(pwd)/depends/x86_64-apple-darwin11
+        make -j2
```
Comment (ultimaweapon) [inline .github/workflows/ci-master.yml]: Same here.

File: .github/workflows/ci-master.yml [hunk 10]
```diff
@@ -38,38 +38,28 @@ jobs:
...
-      run: make -C depends -j2 HOST=x86_64-apple-darwin11
+      run: brew install automake berkeley-db4 libtool boost miniupnpc pkg-config python qt libevent qrencode ccache
     - name: Build Zcoin
       run: |
         ./autogen.sh
-        ./configure --enable-exodus --prefix=$(pwd)/depends/x86_64-apple-darwin11
-        make -j2
\ No newline at end of file
+        ./configure --enable-exodus
+        make -j$(nproc)
```
Comment (ultimaweapon) [inline .github/workflows/ci-master.yml]: Seem like `$(nproc)` is not working on macOS.

Comment (riordant) [reply .github/workflows/ci-master.yml]: correct. you can access `nproc` on MacOS with `sysctl -n hw.physicalcpu`.

**Comment by NorseGaud** [comment]
LOVE IT <3

**Pull request merged**